app/angular/src/client/preview/index.js

```diff
@@ -38,10 +40,10 @@ if (isBrowser) {
 
   // Handle keyboard shortcuts
   window.onkeydown = handleKeyboardShortcuts(channel);
-}
 
-const clientApi = new ClientApi(context);
-export const { storiesOf, setAddon, addDecorator, clearDecorators, getStorybook } = clientApi;
+  // Provide access to external scripts
+  window.__STORYBOOK_CLIENT_API__ = clientApi;
```

Comment:
@tmeasday Is there any particular reason to do that only inside `isBrowser` branch?

Reply:
I guess `window` is not defined in the other context (i.e. jest). Are you thinking we could add it to `global.__STORYBOOK_CLIENT_API__`?

Reply:
I would like to use it from jsdom, where `window` is actually defined and is not the same as `global`. My usecase is to get stories tree from node.js

Reply:
This works in JSDOM. Have you tested it? Storybook doesn't know the difference.

Reply:
I literally tested this yesterday, that's why I feel quite confident about this ;)

Reply:
Did you change default user agent?
https://github.com/storybooks/storybook/blob/c8307fac5bcf61e4f72869c81faa4a3ae05fd56f/app/angular/src/client/preview/index.js#L22

Reply:
I guess I did 😊 

When was that line added? That doesn't seem helpful to me.

Reply:
JSDOM is trying to pretend to be a browser, why would we want to treat it differently?

Reply:
Probably because it lacks api like postmessage that is used in channels

Cc @ndelangen 
https://github.com/storybooks/storybook/pull/2542#discussion_r158586413

Reply:
Hmm, fair enough.

Well I think it is fine for this particular check just be something `if (typeof window !== 'undefined')` but I suspect there will be other use cases for JSDOM where having `isBrowser === true` might be helpful. If it breaks however I guess we leave it.

Reply:
Also any thoughts on https://github.com/storybooks/storybook/pull/3058#issuecomment-374431611